File: wear/src/main/java/cz/johrusk/showsmscode/ShowActivity.java
package cz.johrusk.showsmscode;

import android.app.Activity;
import android.os.Bundle;
import android.os.Vibrator;
import android.support.wearable.view.WatchViewStub;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.TextView;

/**
 * Created by Pepa on 27.05.2016.
 */

public class ShowActivity extends Activity {
    private TextView tv_code;
    private TextView tv_sender;
    static final String TAG = ShowActivity.class.getSimpleName();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Vibrator vibrator = (Vibrator) getSystemService(VIBRATOR_SERVICE);
        long[] vibrationPattern = {0, 500, 50, 300};
        //-1 - don't repeat
        final int indexInPatternToRepeat = -1;
        vibrator.vibrate(vibrationPattern, indexInPatternToRepeat);

        Log.d(TAG,"ShowActivity");
       final String sArray = getIntent().getStringExtra("code");
        final String[] parts = sArray.split("/");

        Log.d(TAG,"Code / Sender: " + parts[0] +" / " + parts[1]);
        setContentView(R.layout.activity_main);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD|WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON);
        final WatchViewStub stub = (WatchViewStub) findViewById(R.id.watch_view_stub);
        stub.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        stub.setOnLayoutInflatedListener(new WatchViewStub.OnLayoutInflatedListener() {
            @Override
            public void onLayoutInflated(WatchViewStub stub) {
                tv_code = (TextView) stub.findViewById(R.id.text);
                tv_code.setText(parts[0]);
                tv_sender = (TextView) stub.findViewById(R.id.sender);
                tv_sender.setText(parts[1]);
            }
        });




    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG,"OnResume");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(TAG,"OnRestart");
    }
}

File: wear/src/main/java/cz/johrusk/showsmscode/ShowActivity.kt
package cz.johrusk.showsmscode

import android.app.Activity
import android.os.Bundle
import android.view.WindowManager
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.AnkoLogger
import org.jetbrains.anko.vibrator
import org.jetbrains.anko.warn

/**
 * Activity which  code and sender of received SMS.

 * @author Josef Hruska (pepa.hruska@gmail.com)
 */

class ShowActivity : Activity(), AnkoLogger {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val vibrationPattern = longArrayOf(0, 500, 50, 300)
        //-1 - don't repeat
        val indexInPatternToRepeat = -1
        vibrator.vibrate(vibrationPattern, indexInPatternToRepeat)

        warn("ShowActivity")
        val sArray = intent.getStringExtra("code")
        val parts = sArray.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        warn("Code / Sender: " + parts[0] + " / " + parts[1])
        setContentView(R.layout.activity_main)
        window.addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD or WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON)

        watch_view_stub.setOnClickListener { finish() }
        watch_view_stub.setOnLayoutInflatedListener { stub ->
            val tv_code = stub.findViewById(R.id.text) as TextView //TODO: how to use Kotlin with ViewStub?
            tv_code.text = parts[0]
            val tv_sender = stub.findViewById(R.id.sender) as TextView
            tv_sender.text = parts[1]
        }
    }

    override fun onResume() {
        super.onResume()
        warn("OnResume")
    }

    override fun onRestart() {
        super.onRestart()
        warn("OnRestart")
    }
}